orbstation/antagonists/traitor/objectives/double_cross.dm

```diff
@@ -0,0 +1,75 @@
+/datum/traitor_objective_category/double_cross
+	name = "Double Cross"
+	objectives = list(
+		/datum/traitor_objective/assassinate/double_cross = 1,
+	)
+	weight = OBJECTIVE_WEIGHT_TINY
+
+// This isn't a subtype of the calling card objective because if I did that
+// I wouldn't be able to make the calling card a different object with different text.
+// Yeah, it's dumb.
+
+/datum/traitor_objective/assassinate/double_cross
+	name = "Assassinate another traitor, and plant a calling card"
+	description = "The Syndicate is looking to get rid of its weakest links. Use your codewords to gain another traitor's trust, asssinate them, and plant a calling card in their pockets."
+
+	progression_minimum = 30 MINUTES
+
+	progression_reward = list(12 MINUTES, 14 MINUTES) // big reward as an incentive
+	telecrystal_reward = list(1, 2) // you can also steal the other traitor's equipment if you want
+
+	maximum_objectives_in_period = 1
+
+	var/obj/item/paper/calling_card/double_cross/card
+
+/datum/traitor_objective/assassinate/double_cross/generate_objective(datum/mind/generating_for, list/possible_duplicates)
+	// This objective will only generate if there is another living traitor
+	var found_traitor = FALSE
+	for(var/datum/mind/possible_target as anything in get_crewmember_minds())
+		if(possible_target == generating_for)
+			continue
+		if(!ishuman(possible_target.current))
+			continue
+		if(possible_target.current.stat == DEAD)
+			continue
+		if(possible_target.has_antag_datum(/datum/antagonist/traitor))
+			found_traitor = TRUE
```

Comment:
Rather than keep this holding variable “found traitor” and returning it at the end of the function, could you just return TRUE here and FALSE if the method exits the loop without returning?  
This would also be more efficient as it can stop looping as soon as it finds one valid target instead of continuing to check everyone else anyway.

Reply:
Oh right, I didn't think of that